ddtrace/internal/ai_guard/api_client.py

```diff
@@ -0,0 +1,170 @@
+"""AI Guard client for security evaluation of agentic AI workflows."""
+import json
+
+import requests
+from typing import Dict, List, Any
+
+import ddtrace.internal.logger as ddlogger
+from ddtrace._trace.tracer import Tracer
+from ddtrace.settings._core import DDConfig
+
+from ddtrace import tracer as ddtracer
+
+logger = ddlogger.get_logger(__name__)
+
+class AIGuardClientError(Exception):
+    """Exception for AI Guard client errors."""
+    pass
+
+
+class AIGuardAbortError(Exception):
+    """Exception to abort current execution due to security policy."""
+    pass
+
+
+class AIGuardWorkflow:
+    """Manages conversation history and evaluates tool calls against AI Guard policies."""
+    def __init__(self, client: "AIGuardClient"):
+        self._client = client
+        self._history: List[Dict[str, Any]] = []
+
+    def add_system_prompt(self, content: str) -> "AIGuardWorkflow":
+        """Add a system prompt to the conversation history."""
+        return self.add_prompt("system", content)
+
+    def add_user_prompt(self, content: str) -> "AIGuardWorkflow":
+        """Add a user prompt to the conversation history."""
+        return self.add_prompt("user", content)
+
+    def add_assistant_prompt(self, content: str) -> "AIGuardWorkflow":
+        """Add an assistant prompt to the conversation history."""
+        return self.add_prompt("assistant", content)
+
+    def add_prompt(self, role: str, content: str) -> "AIGuardWorkflow":
+        """Add a prompt with specified role to the conversation history."""
+        self._history.append({"role": role, "content": content})
+        return self
+
+    def add_tool_output(self, tool: str, args: Dict[str, Any], output: str) -> "AIGuardWorkflow":
+        """Add tool execution results to the conversation history."""
+        self._history.append({"tool": tool, "args": args, "output": output})
+        return self
+
+    def evaluate_tool_call(self, tool: str, args: Any) -> bool:
+        """Evaluate if a tool call is safe to execute.
+        
+        Args:
+            tool: Name of the tool being called
+            args: Arguments being passed to the tool
+            
+        Returns:
+            True if tool execution is allowed, False if denied
+            
+        Raises:
+            AIGuardAbortError: If execution should be aborted
+            AIGuardClientError: If evaluation request fails
+        """
+        attributes = self._client._evaluate(self._history, {"tool_name": tool, "tool_args": args})
+        action = attributes["action"]
+
+        if action == "ALLOW":
+            return True
+        elif action == "DENY":
+            return False
+        else:
```

Comment:
possibly better to handle this explicitly for `ABORT`, and the `else` is an unexpected error?

Reply:
This is controlled inside `_evaluate` in the client:

```python
if action in ["ABORT", "DENY", "ALLOW"]:
                with self._tracer.trace("ai_guard") as span:
                    span.set_tag("ai_guard.action", action)

                return attributes
            else:
                raise AIGuardClientError(f"AI Guard service returned unrecognized action: '{action}'. Expected ALLOW, DENY or ABORT")

```